resources/build-npm-esm.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
```

Comment:
since most of the code is shared with `build-npm.js` it makes sense to just add a cli option to it.
Also please check all places where `build-npm` is used directly or through NPM cli and adjust them accordingly.